### ISSUE TITLE
Add backend jar to artifactory

### ISSUE THREAD

**Issue by geoffjentry** [broadinstitute/cromwell#1019]
Verify that the backend jar is fat, otherwise the components will need to be up there too

**Comment by kshakir** [comment]
**TL;DR Use `sbt publish` to push all the non-fat jars, such as `cromwell-backend >>> _2.11-0.1 <<< .jar`, and do not custom upload the fat `cromwell-backend >>> -0.1 <<< .jar`.**

Via the sbt-assembly plugin [docs](https://github.com/sbt/sbt-assembly/tree/v0.14.1#publishing-not-recommended):

> Publishing fat JARs out to the world is discouraged because non-modular JARs cause much sadness. One might think non-modularity is convenience but it quickly turns into a headache the moment your users step outside of Hello World example code.

The fat jars being generated for our sub-modules should ~~die in a fire~~ be removed via [`aggregate in assembly := false`](http://stackoverflow.com/a/30828390/3320205). Also be sure to clean out the proliferation in Settings.scala of `assemblyJarName in assembly` and the viral `val commonSettings = … ++ assemblySettings ++ …`.

`assemblySettings` and `assemblyJarName` only belong in the `root`!

I have also buried the lede a bit. Our cromwell versioning is... _incomplete_ at the moment, depending on if "Add backend jar" means releases-only or releases-and-snapshots. While we could technically publish releases as is, we shouldn't really publish any snapshots until #645 is fixed, or downstream devs are gonna have a bad time as unhashed snapshots continuously change with each re-publish.

**Comment by geoffjentry** [comment]
We had a meeting yesterday and the following was decided (@kshakir, please double check this)

- All individual jars shall be built non-fat (excluding the final Cromwell executable, which won't go on artifactory) and pushed to artifactory
- Version numbers shall be consistent between all artifacts and not vary independently
